## Wetterdaten data cleaning & filter

test für das säubern und formatieren von Wetterdaten

__Ziel__: Mit einer Schleife auf alle Ordner/csv-Dateien durchführen lassen

In [53]:
import pandas as pd

In [416]:
df_weather = pd.read_csv("Daten/Wetterdaten/tageswerte_KL_05792_19000801_20231231_hist/produkt_klima_tag_19000801_20231231_05792.csv", delimiter=";")
df_weather.head()
# Daten\Wetterdaten\tageswerte_KL_05792_19000801_20231231_hist\produkt_klima_tag_19000801_20231231_05792.csv

STATIONS_ID  MESS_DATUM  QN_3     FX     FM  QN_4    RSK  RSKF   SDK  \
0         5792    19000801  -999 -999.0 -999.0     5 -999.0  -999  15.0   
1         5792    19000802  -999 -999.0 -999.0     5 -999.0  -999   6.2   
2         5792    19000803  -999 -999.0 -999.0     5 -999.0  -999   1.3   
3         5792    19000804  -999 -999.0 -999.0     5 -999.0  -999   0.0   
4         5792    19000805  -999 -999.0 -999.0     5 -999.0  -999   0.0   

   SHK_TAG    NM   VPM     PM   TMK   UPM   TXK   TNK   TGK  eor  
0     -999   0.0   3.9  715.3   2.0  50.0   6.2  -3.6  -999  eor  
1     -999   5.0   8.3  713.7   5.7  88.0  12.6   2.2  -999  eor  
2     -999   8.0   7.9  710.6   3.6  96.0   7.9   1.2  -999  eor  
3     -999   8.0   6.3  706.3   0.6  98.0   5.2  -2.8  -999  eor  
4     -999   8.0   4.4  705.9  -3.9  95.0  -0.9  -5.4  -999  eor

### int zu datetime

In [417]:
df_weather["MESS_DATUM"] = pd.to_datetime(df_weather["MESS_DATUM"], format='%Y%m%d')

In [418]:
df_weather.dtypes

STATIONS_ID             int64
MESS_DATUM     datetime64[ns]
QN_3                    int64
  FX                  float64
  FM                  float64
QN_4                    int64
 RSK                  float64
RSKF                    int64
 SDK                  float64
SHK_TAG                 int64
  NM                  float64
 VPM                  float64
  PM                  float64
 TMK                  float64
 UPM                  float64
 TXK                  float64
 TNK                  float64
 TGK                    int64
eor                    object
dtype: object

### löschen von Spalten
__Spalten die nur das Qualitätsbyte -999 enthalten sollen gelöscht werden__

In [342]:
df_weather.columns.array

<NumpyExtensionArray>
['STATIONS_ID',  'MESS_DATUM',        'QN_3',        '  FX',        '  FM',
        'QN_4',        ' RSK',        'RSKF',        ' SDK',     'SHK_TAG',
        '  NM',        ' VPM',        '  PM',        ' TMK',        ' UPM',
        ' TXK',        ' TNK',        ' TGK',         'eor']
Length: 19, dtype: object

In [434]:
#df_weather["STATIONS_ID"].unique()
#df_weather["MESS_DATUM"].unique()
#df_weather["QN_3"].unique()
#df_weather["  FX"].unique()
#df_weather["  FM"].unique()
#df_weather["QN_4"].unique()
#df_weather[" RSK"].unique()
#df_weather["RSKF"].unique()
#df_weather[" SDK"].unique()
#df_weather["SHK_TAG"].unique()
#df_weather["  NM"].unique()
#df_weather[" VPM"].unique()
#df_weather["  PM"].unique()
#df_weather[" TMK"].unique()
#df_weather[" UPM"].unique()
#df_weather[" TXK"].unique()
#df_weather[" TNK"].unique()
df_weather[" TGK"].unique()
#df_weather["eor"].unique()


array([-999], dtype=int64)

In [435]:
columns_to_drop = ["STATIONS_ID", "eor", " TGK"]
df_weather = df_weather.drop(columns=columns_to_drop)

In [436]:
df_weather.head()

MESS_DATUM  QN_3     FX     FM  QN_4    RSK  RSKF   SDK  SHK_TAG    NM  \
0 1900-08-01  -999 -999.0 -999.0     5 -999.0  -999  15.0     -999   0.0   
1 1900-08-02  -999 -999.0 -999.0     5 -999.0  -999   6.2     -999   5.0   
2 1900-08-03  -999 -999.0 -999.0     5 -999.0  -999   1.3     -999   8.0   
3 1900-08-04  -999 -999.0 -999.0     5 -999.0  -999   0.0     -999   8.0   
4 1900-08-05  -999 -999.0 -999.0     5 -999.0  -999   0.0     -999   8.0   

    VPM     PM   TMK   UPM   TXK   TNK  
0   3.9  715.3   2.0  50.0   6.2  -3.6  
1   8.3  713.7   5.7  88.0  12.6   2.2  
2   7.9  710.6   3.6  96.0   7.9   1.2  
3   6.3  706.3   0.6  98.0   5.2  -2.8  
4   4.4  705.9  -3.9  95.0  -0.9  -5.4

### filter abspeichern

In [437]:
filename = "Daten/cleaned_data/5792_produkt_klima_tag"
df_weather.to_csv(filename, index=False)
# Bestätige die Speicherung der CSV-Datei
import os
print(f"Die CSV-Datei wurde gespeichert: {os.path.abspath(filename)}")

Die CSV-Datei wurde gespeichert: e:\OneDrive\Hochschule\DSC_Semester4\Data Analytics and Business Intelligence 2\Projekt\dabi2-project\Daten\cleaned_data\5792_produkt_klima_tag


### Filter auf alle produkt_klima_tag*-Dateien

### (Je Nach Untersuchungsart nochmal aufteilen)
__zum Beispiel "Wind", "Niederschlag", "Temperatur", ...__